File: python/prism_sdk/throughput_knowledge_representation_contract_model.py
"""Worldgen P04 AFA-worldgen-P04-F07 product wrapper."""
from .worldgen_knowledge_contract_support import KnowledgeContractRequest, KnowledgeContractReceipt, manifest as _manifest, negotiate as _run
FEATURE_ID="AFA-worldgen-P04-F07"
CONTRACT_VERSION="worldgen-throughput-knowledge-contract/1.0"
def worldgen_throughput_knowledge_representation_contract_model_manifest(): return _manifest(feature_id=FEATURE_ID, contract_version=CONTRACT_VERSION, input_schema="KnowledgeContractRequest1@1", scale="prospective high-throughput", autonomy_tier="A1")
def negotiate_worldgen_throughput_knowledge_representation_contract(request): return _run(request, feature_id=FEATURE_ID, contract_version=CONTRACT_VERSION, scale="prospective high-throughput", require_approval=False, require_federation=False)
__all__=["KnowledgeContractRequest","KnowledgeContractReceipt","worldgen_throughput_knowledge_representation_contract_model_manifest","negotiate_worldgen_throughput_knowledge_representation_contract"]

File: crates/worldgen/src/throughput_knowledge_representation_contract_model.rs
//! Worldgen P04-F07 prospective high-throughput knowledge-representation contract model.
use super::knowledge_contract_support::{self, KnowledgeContractRequest, KnowledgeContractReceipt};
pub const FEATURE_ID:&str="AFA-worldgen-P04-F07"; pub const CONTRACT_VERSION:&str="worldgen-throughput-knowledge-contract/1.0";
pub fn worldgen_throughput_knowledge_representation_contract_model_manifest()->serde_json::Value{knowledge_contract_support::manifest(FEATURE_ID,CONTRACT_VERSION,"KnowledgeContractRequest1@1","prospective high-throughput","A1")}
pub fn negotiate_worldgen_throughput_knowledge_contract(request:&KnowledgeContractRequest)->Result<KnowledgeContractReceipt,knowledge_contract_support::KnowledgeContractError>{knowledge_contract_support::negotiate(request,FEATURE_ID,CONTRACT_VERSION,"prospective high-throughput",true)}
pub use knowledge_contract_support::{KnowledgeContractError,KnowledgeContractReceipt as WorldgenThroughputKnowledgeContractReceipt,KnowledgeContractRequest as WorldgenThroughputKnowledgeContractRequest};
